pkg/operations/armhelpers/azureclients.go

```diff
@@ -0,0 +1,32 @@
+package armhelpers
+
+import (
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/Azure/azure-sdk-for-go/arm/resources/resources"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+// AzureClients contains Azure clients
+type AzureClients struct {
+	SubscriptionID   string
+	AzureEnvironment azure.Environment
+	TenantID         string
+
+	GroupsClient *resources.GroupsClient
+	VMClient     *compute.VirtualMachinesClient
+}
+
+// Create method creates various Azure clients
+func (ac *AzureClients) Create(token *adal.ServicePrincipalToken) (*AzureClients, error) {
```

Comment:
I think we should try to get to the point where we're just passing around a fully self-sufficient `AzureClients` struct and not need to keep plumbing subId and token everywhere.